src/main/java/uk/gov/hmcts/sscs/service/SubmitAppealService.java

```diff
@@ -1,72 +1,102 @@
+import uk.gov.hmcts.sscs.exception.CcdException;
 import uk.gov.hmcts.sscs.exception.PdfGenerationException;
+import uk.gov.hmcts.sscs.model.ccd.CaseData;
+import uk.gov.hmcts.sscs.model.ccd.Subscription;
+import uk.gov.hmcts.sscs.model.pdf.PdfWrapper;
+import uk.gov.hmcts.sscs.transform.deserialize.SubmitYourAppealToCcdCaseDataDeserializer;
 
 @Service
 public class SubmitAppealService {
-    private static final Logger LOG = getLogger(SubmitAppealService.class);
 
-    public static final String SYA_CASE_WRAPPER = "SyaCaseWrapper";
-    public static final String ID_FORMAT = "%s_%s";
+    private static final String PDF_WRAPPER = "PdfWrapper";
+    private static final String ID_FORMAT = "%s_%s";
 
-    private String appellantTemplatePath;
+    private final String appellantTemplatePath;
+    private final AppealNumberGenerator appealNumberGenerator;
+    private final SubmitYourAppealToCcdCaseDataDeserializer submitYourAppealToCcdCaseDataDeserializer;
+    private final CcdService ccdService;
     private final PDFServiceClient pdfServiceClient;
     private final EmailService emailService;
-    private SubmitYourAppealEmail submitYourAppealEmail;
+    private final SubmitYourAppealEmail submitYourAppealEmail;
 
     @Autowired
-    public SubmitAppealService(@Value("${appellant.appeal.html.template.path}") String appellantTemplatePath,
-                               PDFServiceClient pdfServiceClient,
-                               EmailService emailService,
-                               SubmitYourAppealEmail submitYourAppealEmail) {
+    SubmitAppealService(@Value("${appellant.appeal.html.template.path}") String appellantTemplatePath,
+                        AppealNumberGenerator appealNumberGenerator,
+                        SubmitYourAppealToCcdCaseDataDeserializer submitYourAppealToCcdCaseDataDeserializer,
+                        CcdService ccdService,
+                        PDFServiceClient pdfServiceClient,
+                        EmailService emailService,
+                        SubmitYourAppealEmail submitYourAppealEmail) {
 
         this.appellantTemplatePath = appellantTemplatePath;
+        this.appealNumberGenerator = appealNumberGenerator;
+        this.submitYourAppealToCcdCaseDataDeserializer = submitYourAppealToCcdCaseDataDeserializer;
+        this.ccdService = ccdService;
         this.pdfServiceClient = pdfServiceClient;
         this.emailService = emailService;
         this.submitYourAppealEmail = submitYourAppealEmail;
     }
 
     public void submitAppeal(SyaCaseWrapper appeal) {
+        CaseData caseData = transformAppealToCaseData(appeal);
+        CaseDetails caseDetails = createCaseInCcd(caseData);
+        byte[] pdf = generatePdf(appeal, caseDetails.getId());
+        sendPdfByEmail(appeal, pdf);
+    }
+
+    private CaseDetails createCaseInCcd(CaseData caseData) {
+        try {
+            return ccdService.createCase(caseData);
+        } catch (CcdException ccdEx) {
```

Comment:
AppInsight alert missing when CCD Exception is thrown.

Reply:
@savirneni , I just configured it as it is for the caseloader which is cleaner btw. And the caseloader is receiving appInsight alerts successfully. The way the code was configured before is not consided best practice. More details here -> https://stackify.com/best-practices-exceptions-java/

Reply:
If we receive alert then thats fine.